op-e2e/withdrawal_helper.go

```diff
@@ -143,35 +162,76 @@ func ProveWithdrawal(t *testing.T, cfg SystemConfig, clients ClientProvider, l2N
 	return params, proveReceipt
 }
 
-func FinalizeWithdrawal(t *testing.T, cfg SystemConfig, l1Client *ethclient.Client, privKey *ecdsa.PrivateKey, withdrawalProofReceipt *types.Receipt, params withdrawals.ProvenWithdrawalParameters) *types.Receipt {
+func FinalizeWithdrawal(t *testing.T, cfg SystemConfig, l1Client *ethclient.Client, privKey *ecdsa.PrivateKey, withdrawalProofReceipt *types.Receipt, params withdrawals.ProvenWithdrawalParameters) (*types.Receipt, *types.Receipt, *types.Receipt) {
 	// Wait for finalization and then create the Finalized Withdrawal Transaction
 	ctx, cancel := context.WithTimeout(context.Background(), 30*time.Duration(cfg.DeployConfig.L1BlockTime)*time.Second)
 	defer cancel()
 
-	err := wait.ForFinalizationPeriod(ctx, l1Client, withdrawalProofReceipt.BlockNumber, config.L1Deployments.L2OutputOracleProxy)
-	require.Nil(t, err)
+	wd := crossdomain.Withdrawal{
+		Nonce:    params.Nonce,
+		Sender:   &params.Sender,
+		Target:   &params.Target,
+		Value:    params.Value,
+		GasLimit: params.GasLimit,
+		Data:     params.Data,
+	}
 
 	opts, err := bind.NewKeyedTransactorWithChainID(privKey, cfg.L1ChainIDBig())
 	require.Nil(t, err)
+
+	var resolveClaimReceipt *types.Receipt
+	var resolveReceipt *types.Receipt
+	if e2eutils.UseFPAC() {
+		portal2, err := bindingspreview.NewOptimismPortal2(config.L1Deployments.OptimismPortalProxy, l1Client)
+		require.Nil(t, err)
+
+		wdHash, err := wd.Hash()
+		require.Nil(t, err)
+
+		game, err := portal2.ProvenWithdrawals(&bind.CallOpts{}, wdHash)
+		require.Nil(t, err)
+		require.NotNil(t, game, "withdrawal should be proven")
+
+		proxy, err := bindings.NewFaultDisputeGame(game.DisputeGameProxy, l1Client)
+		require.Nil(t, err)
+
+		expiry, err := proxy.GameDuration(&bind.CallOpts{})
+		require.Nil(t, err)
+
+		time.Sleep(time.Duration(expiry) * time.Second)
+		resolveClaimTx, err := proxy.ResolveClaim(opts, common.Big0)
+		require.Nil(t, err)
+
+		resolveClaimReceipt, err = geth.WaitForTransaction(resolveClaimTx.Hash(), l1Client, 3*time.Duration(cfg.DeployConfig.L1BlockTime)*time.Second)
```

Comment:
Can we use `wait.ForReceiptOK` instead of `geth.WaitForTransaction` in these tests? `geth.WaitForTransaction` doesn't check the receipt, and will fail on transient network errors.